### PR TITLE
refactor(parser): more strict parser for format_options.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

the problem is described in https://github.com/datafuselabs/databend/pull/9626

 add `format_options`, `connection_options`, `credentails_options` which are more strict.

1. use keyword for option keys
2. use keyword for option values of `type` and `compression`, but allow `string literal ` for compatibility, forbidden `ident` 
3. check string/int/bool for literal values

Closes #issue

### CODE REVIEW

File: src/query/ast/src/parser/stage.rs [hunk 1]
```diff
@@ -36,11 +36,38 @@ pub fn u64_to_string(i: Input) -> IResult<String> {
...
+    let option_compression = map(
+        rule! {
+        (COMPRESSION ~ "=" ~ #parameter_to_string )
+        },
+        |(_, _, v)| ("COMPRESSION".to_string(), v.clone()),
+    );
+
+    let string_options = map(
+        rule! {
+            (TYPE | RECORD_DELIMITER | FIELD_DELIMITER | QUOTE | SKIP_HEADER | NON_DISPLAY | ESCAPE ) ~ "=" ~ #literal_string
```
Comment (sundy-li) [inline src/query/ast/src/parser/stage.rs:60]: What's QUOTE here ？

Comment (youngsofun) [reply src/query/ast/src/parser/stage.rs:60]: a csv option

**Comment by youngsofun** [comment]
```
 // TODO: We will use `CONNECTION` to replace `CREDENTIALS` and `ENCRYPTION`.
```

is it time to do it?

ENCRYPTION not  seen in doc,  only in unit test 

```
      ENCRYPTION = (
                    MASTER_KEY = 'master_key'
                )
```

CREDENTIALS is used in doc

https://databend.rs/doc/sql-commands/dml/dml-copy-into-location#externallocation-for-amazon-s3

**Comment by Xuanwo** [comment]
Let's do this!

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
```

Differences (-left|+right):
 error: 
   --> SQL:1:38
   |
 1 | COPY INTO mytable FROM 's3://bucket' CREDENTIAL = ();
-  |                                      ^^^^^^^^^^ expected `CONNECTION`, `CREDENTIALS`, `LOCATION_PREFIX`, `FILES`, `PATTERN`,`FILE_FORMAT`, or 10 more ...
+  |                                      ^^^^^^^^^^ expected `CONNECTION`, `CREDENTIALS`, `LOCATION_PREFIX`, `FILES`, `PATTERN`, `FILE_FORMAT`, or 10 more ...
```

**Pull request merged**